lib/spack/spack/environment/environment.py

```diff
@@ -2731,8 +2731,8 @@ def included_config_scopes(self) -> List[spack.config.ConfigScope]:
 
             include_url = urllib.parse.urlparse(config_path)
 
-            # Transform file:// URLs to direct includes.
-            if include_url.scheme == "file":
+            # Transform file:// or C:// URLs to direct includes.
+            if include_url.scheme in ("file", "c"):
```

Comment:
I think this is fine for CI, but I think any single letter is allowed, so this should include c-z.

Reply:
I think "c" being interpreted as the scheme is technically a bug, and ideally we'd wrap urlparse with something that can understand this on Windows.

I think a TODO would be helpful, if we don't change that here.

Reply:
IIRC we tried to fix this last summer and ended up uncovering a bit of a hornet's nest w.r.t. url parsing on Windows. I'll check into Markus' old PRs and see if I can pull some of that effort in here, otherwise will genericize using our drive letter regex.

Reply:
I don't want to overextend this, so I'll say you can take a partial approach if you like (or add a TODO).

Reply:
I was thinking of Harmen's PR https://github.com/spack/spack/pull/34452
The helper methods in here should fix this since we always expect this to be a file url.

Reply:
On second thought, I think this indicates a more fundamental problem. Why are we passing valid file urls in one context and raw Windows paths in another? The Windows paths should also be formatted as file urls so they can be picked up properly by url parse. If we were just passing raw paths on *nix platforms, we'd get `''` which would also fail here.

Going to look into where this is coming from and fix it there.

Reply:
This has been handled by relying on scheme validation rather than a hard check for Windows drive labels.